### PR TITLE
Added the edge test cases.

### LINKED ISSUE #230: Dependancy issue in Fast Sync Mode
## Context

When restarting the bridge service by reseting DB, it will try to sync from scratch in both networks. 
`WrappedToken` table is dependent on `Deposit` table, so when it detects `WrappedToken` event, it will refer to the deposit table of the opposite network. It means `WrappedToken` event comes up after `AddDeposit` event of the opposite network. In the real env, it works. But in the fast sync mode, each goroutine are trying to sync events independently.

## Suggestion

```sql
CREATE TABLE syncv2.deposit
(
    network_id  INTEGER,
    orig_net    INTEGER,
    token_addr  BYTEA NOT NULL,
    amount      VARCHAR,
    dest_net    INTEGER NOT NULL,
    dest_addr   BYTEA NOT NULL,
    block_id    BIGINT NOT NULL REFERENCES syncv2.block (id) ON DELETE CASCADE,
    deposit_cnt BIGINT,
    tx_hash     BYTEA NOT NULL,
    metadata    BYTEA NOT NULL,
    PRIMARY KEY (network_id, deposit_cnt)
);

CREATE TABLE syncv2.token_wrapped
(
    network_id         INTEGER NOT NULL,
    orig_net           INTEGER,
    orig_token_addr    BYTEA NOT NULL,
    wrapped_token_addr BYTEA NOT NULL,
    block_id           BIGINT NOT NULL REFERENCES syncv2.block (id) ON DELETE CASCADE,
    name               VARCHAR,
    symbol             VARCHAR,
    decimals           INTEGER
);
```
As see, the fields of `token_wrapped` like `name, symbol, decimals` are dependent on `metadata` of `deposit` table.
We will ignore these fields when can't find the related deposit data (this case should be a fast sync mode).
And we will try to update the `token_wrapped` table when facing with `getTokenWrapped` API. (it will be a normal mode)

### PR DESCRIPTION
**Pull request by cool-develope**
Closes #230 
ref #217 

### What does this PR do?

- Added more test cases to simulate when restarts the service without db resets and detects L1 reorgs.

### Reviewers

Main reviewers:

- @ARR552 
- @tclemos 

Codeowner reviewers:

- @arnaubennassar

### CODE REVIEW

File: db/pgstorage/pgstorage.go [hunk 1]
```diff
@@ -349,11 +356,33 @@ func (p *PostgresStorage) GetLatestTrustedExitRoot(ctx context.Context, dbTx pgx
...
 func (p *PostgresStorage) GetTokenWrapped(ctx context.Context, originalNetwork uint, originalTokenAddress common.Address, dbTx pgx.Tx) (*etherman.TokenWrapped, error) {
 	const getWrappedTokenSQL = "SELECT network_id, orig_net, orig_token_addr, wrapped_token_addr, block_id, name, symbol, decimals FROM syncv2.token_wrapped WHERE orig_net = $1 AND orig_token_addr = $2"
+
 	var token etherman.TokenWrapped
 	err := p.getExecQuerier(dbTx).QueryRow(ctx, getWrappedTokenSQL, originalNetwork, originalTokenAddress).Scan(&token.NetworkID, &token.OriginalNetwork, &token.OriginalTokenAddress, &token.WrappedTokenAddress, &token.BlockID, &token.Name, &token.Symbol, &token.Decimals)
 	if errors.Is(err, pgx.ErrNoRows) {
 		return nil, gerror.ErrStorageNotFound
 	}
+
+	if token.Symbol == "" { // this is due to missing the related deposit in the opposite network in fast sync mode.
```
Comment (ARR552) [inline db/pgstorage/pgstorage.go:366]: Can you give more details here?

File: db/pgstorage/pgstorage.go [hunk 2]
```diff
@@ -505,3 +534,10 @@ func (p *PostgresStorage) ResetTrustedState(ctx context.Context, batchNumber uin
 	_, err := p.getExecQuerier(dbTx).Exec(ctx, resetTrustedStateSQL, batchNumber)
 	return err
 }
+
+// UpdateBlocks updates the hash of blocks.
+func (p *PostgresStorage) UpdateBlocks(ctx context.Context, networkID uint, blockNum uint64, dbTx pgx.Tx) error {
```
Comment (ARR552) [inline db/pgstorage/pgstorage.go:539]: Is this function used only for testing purposes right? Maybe we coan call it UpdateBlocksForTesting or something like that

File: test/e2e/edge_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	const st time.Duration = 10 // wait unitil the syncing is finished
+	time.Sleep(st * time.Second)
+
+	t.Run("Test a case of restart with reorg.", func(t *testing.T) {
+		depositFromL1(ctx, opsman, t)
+		// Modify the L1 blocks
+		destAddr := common.HexToAddress("0xc949254d682d8c9ad5682521675b8f43b102aec4")
+		deposits, err := opsman.GetBridgeInfoByDestAddr(ctx, &destAddr)
+		require.NoError(t, err)
+		require.NoError(t, opsman.UpdateBlocks(ctx, 0, deposits[0].BlockNum))
```
Comment (ARR552) [inline test/e2e/edge_test.go:90]: Are you trying to simulate a reorg here?

Comment (cool-develope) [reply test/e2e/edge_test.go:90]: yes

File: bridgectrl/merkletree.go [hunk 4]
```diff
@@ -72,7 +73,7 @@ func (mt *MerkleTree) getSiblings(ctx context.Context, index uint, root [KeyLen]
 	for h := mt.height - 1; ; h-- {
 		value, err := mt.store.Get(ctx, cur[:], nil)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("height: %d, cur: %v, error: %v", h, cur, err)
```
Comment (arnaubennassar) [inline bridgectrl/merkletree.go:76]:
```suggestion
			return nil, fmt.Errorf("height: %d, cur: %v, error: %w", h, cur, err)
```

**Review by arnaubennassar — CHANGES_REQUESTED** [review]

**Review by arnaubennassar — APPROVED** [review]

**Pull request merged**